aperturedb/__init__.py

```diff
@@ -47,13 +47,30 @@
         os.environ["ADB_LOG_FILE"]) == 0 else os.environ["ADB_LOG_FILE"]
 
 if error_file_name is not None:
+    # detachable file handler allows program to run from working directory
+    # without write access; adb is a use case for this.
+    class DetachableFileHandler(logging.FileHandler):
+        def __init__(self, file_path, delay=False):
+            super().__init__(file_path, delay=delay)
+
+        def emit(self, record):
+            try:
+                super().emit(record)
+            except PermissionError as pErr:
+                for h in logger.handlers[:]:
+                    if h == self:
+                        logger.removeHandler(h)
+                        break
+                logging.warning(
```

Comment:
This log level could be bumped up, so that the line makes it to the console. That might prove useful.